### PR TITLE
[WIP] {docs, session}: add consecutive user message handling to session management

### PR DESCRIPTION
**Pull request by Rememorio**
- Introduced a new section in the documentation detailing the handling of consecutive user messages in session management.
- Implemented an `OnDuplicateUserMessage` callback mechanism to manage scenarios where a user sends multiple messages without receiving a response, preventing errors from LLM APIs.
- Provided examples of common handler implementations, including inserting placeholder messages, removing previous messages, and skipping current messages.
- Added a custom handler example for merging consecutive user messages.
- Updated session management code to support this feature, ensuring backward compatibility and session-level configuration options.

## Summary by Sourcery

在会话管理中新增可配置的连续用户消息处理机制，以防止生成无效的历史记录并避免 LLM API 错误。

新功能：
- 在会话上引入 `OnDuplicateUserMessage` 回调，用于自定义如何处理连续的用户消息。
- 增加会话选项的连接配置，使处理器可以按会话进行配置，并在会话更新期间自动调用。
- 提供一个可运行的示例程序，演示在多个会话后端中使用不同的重复用户消息处理策略。

文档：
- 在英文和中文的会话管理指南中记录连续用户消息的处理方式，包括内置处理器和自定义处理器模式。
- 在 README 中为 `OnDuplicateUserMessage` 示例新增文档，解释用法、使用场景以及处理器行为。

测试：
- 为重复用户消息处理器添加全面测试，覆盖内置策略、自定义合并、正常流程、多条连续消息、无处理器情况下的行为以及并发行为。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add configurable handling for consecutive user messages in session management to prevent invalid histories and LLM API errors.

New Features:
- Introduce an OnDuplicateUserMessage callback on sessions to customize how consecutive user messages are resolved.
- Add session option wiring so handlers can be configured per session and invoked automatically during session updates.
- Provide a runnable example program demonstrating different duplicate-user-message strategies across multiple session backends.

Documentation:
- Document consecutive user message handling in both English and Chinese session management guides, including built-in and custom handler patterns.
- Add README documentation for the OnDuplicateUserMessage example explaining usage, scenarios, and handler behaviors.

Tests:
- Add comprehensive tests for the duplicate user message handler covering built-in strategies, custom merging, normal flows, multiple consecutive messages, absence of handlers, and concurrency behavior.

</details>

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

在会话管理中新增可配置的连续用户消息处理机制，以防止生成无效的历史记录并避免 LLM API 错误。

新功能：
- 在会话上引入 `OnDuplicateUserMessage` 回调，用于自定义如何处理连续的用户消息。
- 增加会话选项的连接配置，使处理器可以按会话进行配置，并在会话更新期间自动调用。
- 提供一个可运行的示例程序，演示在多个会话后端中使用不同的重复用户消息处理策略。

文档：
- 在英文和中文的会话管理指南中记录连续用户消息的处理方式，包括内置处理器和自定义处理器模式。
- 在 README 中为 `OnDuplicateUserMessage` 示例新增文档，解释用法、使用场景以及处理器行为。

测试：
- 为重复用户消息处理器添加全面测试，覆盖内置策略、自定义合并、正常流程、多条连续消息、无处理器情况下的行为以及并发行为。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add configurable handling for consecutive user messages in session management to prevent invalid histories and LLM API errors.

New Features:
- Introduce an OnDuplicateUserMessage callback on sessions to customize how consecutive user messages are resolved.
- Add session option wiring so handlers can be configured per session and invoked automatically during session updates.
- Provide a runnable example program demonstrating different duplicate-user-message strategies across multiple session backends.

Documentation:
- Document consecutive user message handling in both English and Chinese session management guides, including built-in and custom handler patterns.
- Add README documentation for the OnDuplicateUserMessage example explaining usage, scenarios, and handler behaviors.

Tests:
- Add comprehensive tests for the duplicate user message handler covering built-in strategies, custom merging, normal flows, multiple consecutive messages, absence of handlers, and concurrency behavior.

</details>

</details>

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -381,14 +418,46 @@ func (sess *Session) EnsureEventStartWithUser() {
...
 		return
 	}
 	if event.Response != nil && !event.IsPartial && event.IsValidContent() {
 		sess.EventMu.Lock()
-		sess.Events = append(sess.Events, *event)
+
+		// Check for consecutive user messages before appending.
+		shouldAppend := true
+		if event.Response.IsUserMessage() && len(sess.Events) > 0 {
+			lastEvent := sess.Events[len(sess.Events)-1]
```
Comment (Flash-LHR) [inline session/session.go]: 这里的 `lastEvent` 是拷贝，handler 收到的 `previousUserEvent` 修改不会写回 `sess.Events`。建议改传 `&sess.Events[len(sess.Events)-1]`。

File: session/session.go [hunk 2]
```diff
@@ -381,14 +418,46 @@ func (sess *Session) EnsureEventStartWithUser() {
...
+			if lastEvent.Response != nil && lastEvent.Response.IsUserMessage() {
+				// Consecutive user messages detected.
+				sess.handlerMu.RLock()
+				handler := sess.onDuplicateUserMsg
+				sess.handlerMu.RUnlock()
+
+				if handler != nil {
+					// Let user handle it.
+					shouldAppend = handler(sess, &lastEvent, event)
```
Comment (Flash-LHR) [inline session/session.go]: handler 在持有 `EventMu` 时被调用，回调里调用 `UpdateUserSession` 会死锁。建议解锁后回调或在文档里明确禁止再入。

Comment (Rememorio) [reply session/session.go]: fixed, 文档说明

File: session/session.go (hunk 1 above)
Comment (Rememorio) [reply session/session.go]: fixed, 传了指针

**Comment by Rememorio** [comment]
这个设计，用 `AppendEventHook` 也能做， 和用户沟通的结论为：改成提供一个 example: https://github.com/trpc-group/trpc-agent-go/pull/1085
